tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py

```diff
@@ -121,19 +127,24 @@ def test_download_feeds(configuration, metadata, set_wazuh_configuration_vdt, tr
         - r'Starting .* database update'
         - r'The update of the .* feed finished successfully'
     '''
-    # Check that the feed update has started
-    evm.check_provider_database_update_start_log(metadata['provider_name'])
-
-    # Check that the feed has been updated successfully
-    evm.check_provider_database_update_finish_log(provider_name=metadata['provider_name'],
-                                                  timeout=metadata['download_timeout'])
-
-    if 'provider_json_name' in metadata:
-        evm.check_provider_database_update_start_log(metadata['provider_json_name'])
-        evm.check_provider_database_update_finish_log(provider_name=metadata['provider_json_name'],
+    try:
+        # Check that the feed update has started
+        evm.check_provider_database_update_start_log(metadata['provider_name'])
+        # Check that the feed has been updated successfully
+        evm.check_provider_database_update_finish_log(provider_name=metadata['provider_name'],
                                                       timeout=metadata['download_timeout'])
 
-    # Check that the timestamp of the feed metadata does not exceed the established threshold limit.
-    assert vd.feed_is_recently_updated(provider_name=metadata['provider_name'], provider_os=metadata['provider_os'],
-                                       threshold_weeks=metadata['update_treshold_weeks']), '' \
-           f"The {metadata['provider_os']} feed has not been recently updated"
+        if 'provider_json_name' in metadata:
+            evm.check_provider_database_update_start_log(metadata['provider_json_name'])
+            evm.check_provider_database_update_finish_log(provider_name=metadata['provider_json_name'],
+                                                          timeout=metadata['download_timeout'])
+
+        # Check that the timestamp of the feed metadata does not exceed the established threshold limit.
+        if metadata['update_treshold_weeks'] != 'None':    
+            assert vd.feed_is_recently_updated(provider_name=metadata['provider_name'],
+                                               provider_os=metadata['provider_os'],
+                                               threshold_weeks=metadata['update_treshold_weeks']), '' \
+                                               f"The {metadata['provider_os']} feed has not been recently updated"
+    except TimeoutError as e:
```

Comment:
I think this is not correct. You are catching all `TimeoutError`, so if a test fails due to timeout, it would pass in case it is not Ubuntu.

I would directly not catch any exception and put the `if` to check if it is ubuntu and put the `xfail` at the beginning.

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/pull/2847/commits/2a8b23b81b8c7074685b72d1653a1bd171f1aaf7

Moving the xfail to the beggining would make the test fail without running anything. If we handle it at the end, we can check if the test passes or fails actually. Added `pytest.fail` for cases that are not `bionic`